src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -269,12 +141,16 @@ public int hashCode() {
      */
     @Override
     public boolean equals(final Object other) {
-        if (! (other instanceof FeatureInput)) {
+        if (other == this){
+            return true;
+        }
+        if (!(other instanceof FeatureInput)){
             return false;
         }
-
-        final FeatureInput<?> otherFeature = (FeatureInput<?>)other;
-        return name.equals(otherFeature.name) && featureFile.equals(otherFeature.featureFile);
+        @SuppressWarnings("rawtypes") //for the cast
+        final FeatureInput that = ((FeatureInput)other);
```

Comment:
the cast is still going to be unchecked. i think SuppressWarnings will need to stay